app/services/state_file/ty24_archiver_service.rb

```diff
@@ -0,0 +1,111 @@
+module StateFile
+  class Ty24ArchiverService
+    INTAKE_MAP = {
+      :az => StateFileAzIntake,
+      :id => StateFileIdIntake,
+      :md => StateFileMdIntake,
+      :nj => StateFileNjIntake,
+      :nc => StateFileNcIntake,
+    }.freeze
+
+    attr_reader :state_code, :batch_size, :intake_class, :tax_year
+
+    def initialize(state_code:, batch_size:)
+      @state_code = state_code
+      @intake_class = INTAKE_MAP[state_code.to_sym]
+      @intake_class = INTAKE_MAP[state_code.to_sym] ||
+        raise(ArgumentError, "Invalid state_code: #{state_code}. Must be one of: #{INTAKE_MAP.keys.join(', ')}")
+      @tax_year = 2024
+      @batch_size = batch_size
+    end
+
+    def self.archive!(state_code:, batch_size: 100)
+      new(state_code: state_code, batch_size: batch_size).archive_all
+    end
+
+    def archive_all
+      Rails.logger.info("*****Archiving for state: #{state_code}, tax_year: {tax_year}, batch_size: #{batch_size}*****")
+      loop do
+        batch = fetch_batch
+        break if batch.empty?
+
+        archived_ids = archive_batch(batch)
+        Rails.logger.info("---Archived #{archived_ids.count} records for #{state_code.upcase}: [#{archived_ids.join(', ')}]---")
+      end
+      Rails.logger.info("*****Completed archive for state: #{state_code}*****")
+    end
+
+    def archive_all
+      Rails.logger.info("*****Archiving for state: #{state_code}, tax_year: #{tax_year}, batch_size: #{batch_size}*****")
+
+      et = Time.find_zone('America/New_York')
+      start_date = et.parse('2025-01-15 00:00:00') # state_file_start_of_open_intake
+      end_date = et.parse('2025-10-25 23:59:59') # state_file_end_of_in_progress_intakes
+
+      # find state file intakes that have been accepted during the season
+      accepted_intake_ids = EfileSubmissionTransition.joins(:efile_submission)
+                                                     .where(efile_submissions: { data_source_type: intake_class.name })
+                                                     .where(
+                                                       efile_submission_transitions: {
+                                                         created_at: start_date..end_date,
+                                                         to_state: :accepted,
+                                                         most_recent: true
+                                                       }
+                                                     ).pluck('efile_submissions.data_source_id')
+
+      # remove intakes that have already been archived
+      archived_emails = StateFileArchivedIntake.where(state_code: state_code, tax_year: tax_year)
+                                               .pluck(:email_address)
+      archived_phones = StateFileArchivedIntake.where(state_code: state_code, tax_year: tax_year)
```

Comment:
why query by the email and phone as opposed to the hashed ssn?

Reply:
i guess they are both unique so it doesn’t really matter but i was just curious

Reply:
Since this is unique identifier that they use to sign in with, they either pick email or phone number. Last year they only did this check on emails since we didn't have sms working in the previous season before that. But actually I was thinking about it more and I don't know why we didn't just add a key reference for that state file intake like `data_source_id` instead wdyt